### DISCUSSION TITLE
`.helix/ignore` should also mask out files from diagnostic pickers

### DISCUSSION THREAD

**Discussion by gamma-delta** [helix-editor/helix#8983] [Ideas]
Hello,

I'm using the `ignore` file feature for Godot/C# development and it's really handy!

But, when I go `space D` to search the workspace for errors, it also displays errors in autogenerated C# files in `.godot/...`, even though I have `.godot/**` in my ignore file.

I think it would make sense if this picker also obeyed the ignore file.

**Comment by gamma-delta** [comment]
OK, I've done a little bit of digging

[The diagnostic picker is defined here](https://github.com/helix-editor/helix/blob/b023faacf898e2f393fa8f7f91e685547080442c/helix-term/src/commands/lsp.rs#L263). It should be possible to use the mechanism for [ignoring files out of the normal file picker](https://github.com/helix-editor/helix/blob/master/helix-term/src/ui/mod.rs#L180) to filter the diagnostics...

**Comment by gamma-delta** [comment]
Looks like it's going to be hard to get perfect feature parity because the `WalkBuilder` does a lot of work behind the scenes to find every single ignore file (searching `~/.ignore`, stuff like that).

I'll just use `<helix-config>/.ignore` and `./.helix/ignore` for now...

**Comment by rcorre** [comment]
Typically ignoring files for diagnostics is done at the language server level. Does the C# language server have a way to ignore files?
